src/Nncase.Tests/Evaluator/UnitTestEvaluatorMath.cs

```diff
@@ -0,0 +1,1033 @@
+        float initValue = 0F;
+        long keepDims = 1;
+        var a = new float[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var result = new float[] { 5, 6, 7, 8 };
+        var ort_a = OrtKISharp.Tensor.MakeTensor(a, new long[] { 2, 4 });
+        var expr_a = Tensor.From(a, new[] { 2, 4 });
+        var expect = Tensor.From(result, new[] { 1, 4 }).ToOrtTensor();
+
+        var ops = new ReduceOp[] { ReduceOp.Max, ReduceOp.Min, ReduceOp.Mean, ReduceOp.Prod, ReduceOp.Sum };
+
+        foreach (var op in ops)
+        {
+            switch (op)
+            {
+                case ReduceOp.Max:
+                    {
+                        expect = OrtKI.ReduceMax(ort_a, axes, keepDims);
+                        break;
+                    }
+
+                case ReduceOp.Min:
+                    {
+                        expect = OrtKI.ReduceMin(ort_a, axes, keepDims);
+                        break;
+                    }
+
+                case ReduceOp.Mean:
+                    {
+                        expect = OrtKI.ReduceMean(ort_a, axes, keepDims);
+                        break;
+                    }
+
+                case ReduceOp.Prod:
+                    {
+                        expect = OrtKI.ReduceProd(ort_a, axes, keepDims);
+                        break;
+                    }
+
+                case ReduceOp.Sum:
+                    {
+                        expect = OrtKI.ReduceSum(ort_a, axes, keepDims, 0L);
+                        break;
+                    }
+            }
+
+            var expr = IR.F.Tensors.Reduce(op, expr_a, Tensor.From(axes, new[] { 1 }), initValue, keepDims);
+            CompilerServices.InferenceType(expr);
+            Assert.Equal(expect, expr.Evaluate().AsTensor().ToOrtTensor());
+        }
+    }
+
+    [Fact]
+    public void TestReduceArg()
+    {
+        long axis = 0L;
+        long[] keepDims = { 0L, 1L };
+        long select_last_idx = 0L;
+        var a = new float[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var result = new int[] { 5, 6, 7, 8 };
+        var ort_a = OrtKISharp.Tensor.MakeTensor(a, new long[] { 2, 4 });
+        var expr_a = Tensor.From(a, new[] { 2, 4 });
+        var expect = Tensor.From(result, new[] { 1, 4 }).ToOrtTensor();
+
+        var ops = new ReduceArgOp[] { ReduceArgOp.ArgMax, ReduceArgOp.ArgMin };
+
+        foreach (var keepdims in keepDims)
+        {
+            foreach (var op in ops)
+            {
+                switch (op)
+                {
+                    case ReduceArgOp.ArgMax:
+                        {
+                            expect = OrtKI.ArgMax(ort_a, axis, keepdims, select_last_idx);
+                            break;
+                        }
+
+                    case ReduceArgOp.ArgMin:
+                        {
+                            expect = OrtKI.ArgMin(ort_a, axis, keepdims, select_last_idx);
+                            break;
+                        }
+                }
```

Comment:
It's more elegant to use [switch expression](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression).